include/networkit/community/LouvainMapEquation.hpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * LouvainMapEquation.hpp
+ *
+ * Created on: 2019-01-28
+ * Author: Armin Wiebigke
+ *         Michael Hamann
+ *         Lars Gottesbüren
+ */
+// networkit-format
+
+#ifndef NETWORKIT_COMMUNITY_LOUVAIN_MAP_EQUATION_HPP_
+#define NETWORKIT_COMMUNITY_LOUVAIN_MAP_EQUATION_HPP_
+
+#include <networkit/auxiliary/SparseVector.hpp>
+#include <networkit/auxiliary/SpinLock.hpp>
+#include <networkit/community/CommunityDetectionAlgorithm.hpp>
+
+namespace NetworKit {
+
+class LouvainMapEquation : public CommunityDetectionAlgorithm {
+private:
+    enum class ParallelizationType : uint8_t { None, RelaxMap, Synchronous };
+
+    /**
+     * @param[in] G input graph
+     * @param[in] hierarchical use recursive coarsening
+     * @param[in] maxIterations maximum number of iterations for move phase
+     * @param[in] parallelization strategy (default synchronous):
+     *            none
+     *            relaxmap    -> one lock per community to update cuts
+     *            synchronous -> work on stale cuts and volumes, update in second step
+     *
+     */
+    explicit LouvainMapEquation(const Graph &graph, bool hierarchical, count maxIterations,
+                                ParallelizationType parallelizationType);
+
+public:
+    /**
+     * @param[in] G input graph
+     * @param[in] hierarchical use recursive coarsening
+     * @param[in] maxIterations maximum number of iterations for move phase
+     * @param[in] parallelization strategy (default synchronous):
+     *            none
+     *            relaxmap    -> one lock per community to update cuts
+     *            synchronous -> work on stale cuts and volumes, update in second step
+     *
+     */
+    explicit LouvainMapEquation(const Graph &graph, bool hierarchical = false,
+                                count maxIterations = 32,
+                                const std::string parallelizationStrategy = "relaxmap")
```

Comment:
CI (clang-tidy checks) complains about this: `the const qualified parameter 'parallelizationStrategy' is copied for each invocation; consider making it a reference [performance-unnecessary-value-param,-warnings-as-errors]`

It would be good to also fix the narrowing-conversion warnings in the CI-log (at least in the files you added). However they only trigger warnings as of now - so it is not neccessary.

Reply:
I'm already on it. Is this check new and how does it not fail on the main branch? I rebased on the most recent version.